addons/point_of_sale/models/pos_order.py

```diff
@@ -611,6 +614,186 @@ def _prepare_invoice_vals(self):
         if self.note:
             vals.update({'narration': self.note})
         return vals
+
+    def _prepare_aml_values_list_per_nature(self):
+        self.ensure_one()
+        sign = 1 if self.amount_total < 0 else -1
+        commercial_partner = self.partner_id.commercial_partner_id
+        company_currency = self.company_id.currency_id
+        rate = self.currency_id._get_conversion_rate(self.currency_id, company_currency, self.company_id, self.date_order)
+
+        # Concert each order line to a dictionary containing business values. Also, prepare for taxes computation.
+        base_line_vals_list = []
+        for line in self.lines.with_company(self.company_id):
+            account = line.product_id._get_product_accounts()['income']
+            if not account:
+                raise UserError(_(
```

Comment:
would a RedirectWarning make sense here? To go to the product directly instead of giving the id and let the user figure it out?

Reply:
Yep totally, nice catch thanks!

Reply:
Hence my question. If it doesn't make sense (e.g. the user seeing the error doesn't necessary have the rights to do the modification), no need to change it :)